tests/containers/validate_btrfs.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Among these features are block-level operations, thin provisioning, copy-on-write snapshots,
+# and ease of administration. You can easily combine multiple physical block devices into a single Btrfs filesystem.
+# The scenario illustrates the build of a sle image and then reuse the same Dockerfile to check that
+# the thin partitioning are used. Then we check the block subvolumes.
+# Finally we test the disk administration filling up the subvolume mounted for the images
+# and try to pull another image. With btrfs we should be able to add another disk and be able to
+# continue when the docker partition is fulled up.
+# Maintainer: qac team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use containers::runtime;
+
+sub run {
+    my $docker = containers::runtime->new('docker');
+    my $dockerfile_path = '/root/sle_base_image/docker_build';
+    $docker->build($dockerfile_path, 'huge_image');
+    assert_script_run "btrfs fi df /var/lib/docker";
+    validate_script_output "df -h |grep var", sub { m/\/dev\/vda.+[1-6]\d?%/ };
+    $docker->up('huge_image', 'fallocate -l 8.7G bigfile.txt');
+    #it should be full
+    validate_script_output "df -h |grep var", sub { m/\/dev\/vda.+(9[7-9]|100)%/ };
+    assert_script_run "btrfs fi df /var/lib/docker";
+    
+    # thin partitioning
+    $docker->build($dockerfile_path, 'thin_image');
+    # todo: check_size
+    # Due to disk space this should fail
+    assert_script_run qq(docker pull fedora || echo pull_failed);
+    assert_script_run "btrfs device add /dev/vdb /var/lib/docker";
+    assert_script_run "df -h ";
+    validate_script_output "lsblk |grep vdb", sub { m/vdb.+20G/ };
+    validate_script_output "btrfs fi show /var/lib/docker/btrfs", sub { m/devid\s+2\s+size\s20.00GiB\sused\s0.00B.+\/dev\/vdb/ };
+    $docker->pull('fedora');
```

Comment:
Why not Tumbleweed and/or Leap? :slightly_smiling_face:

Reply:
it could be anything here. it doesnt really matter. but yup i think one of the suggestion is more appropriate here.thanks